cosmotheka/mappers/mapper_ROSAT.py

```diff
@@ -47,17 +47,39 @@ def __init__(self, config):
         self.rot = self._get_rotator('C')
         self.fname_expmap = config['exposure_map']
         self.fname_pholist = config['photon_list']
+        self.fname_pscat = config['point_source_catalog']
         self.erange = config.get('energy_range', [0.5, 3.0])
         self.explimit = config.get('exposure_min', 100.0)
         self.mask_external = config.get('external_mask', None)
-        self.npix = hp.nside2npix(self.nside)
+        self.mask_ps = config.get('mask_point_sources', False)
 
+        # Nside for point source masking. This is fixed because the
+        # corresponding pixel size is similar to the ROSAT PSF.
+        self.nside_ps = 2048
+        self.count_map = None
         self.expmap = None
         self.pholist = None
         # signal_map is a map of the countrate
         self.nl_coupled = None
+        self.ps_mask = None
 
-    def get_pholist(self):
+    def _get_ps_mask(self):
+        """ Returns point-source mask at nside=2048
+        """
+        if self.ps_mask is None:
+            f = fitsio.FITS(self.fname_pscat)
+            cat = f[1].read()
+            # Flux cut
+            cat = cat[cat['RATE'] >
+                      self.config['point_source_flux_cut_ctps']]
+            nmap = get_map_from_points(cat, self.nside_ps,
+                                       ra_name='RA_DEG',
+                                       dec_name='DEC_DEG',
+                                       rot=self.rot)
+            self.ps_mask = (nmap == 0).astype(float)
+        return self.ps_mask
+
+    def _get_pholist(self):
```

Comment:
Maybe we want to have the catalog methods public? Sth to discuss in all mappers, not only this one.

Reply:
the less we make public, the less we have to document and maintain. But yes, we can discuss outside of this.

Reply:
Yeah. I think we can think of a way of unifying all the catalog-based mappers so they share a public `get_catalog` method. That way we just document it once. 
Future work, anyway.